src/anomalib/models/image/reverse_distillation/anomaly_map.py

```diff
@@ -87,8 +88,7 @@ def forward(self, student_features: list[torch.Tensor], teacher_features: list[t
             elif self.mode == AnomalyMapGenerationMode.ADD:
                 anomaly_map += distance_map
 
-        return gaussian_blur2d(
-            anomaly_map,
+        return GaussianBlur2d(
             kernel_size=(self.kernel_size, self.kernel_size),
             sigma=(self.sigma, self.sigma),
-        )
+        )(anomaly_map.to("cpu"))
```

Comment:
can you elaborate this needs to be explicitly moved to the cpu? Would it be an idea to check the device based on the input?

Reply:
Yep. That's a better idea